### DISCUSSION TITLE
Best way to sync storage bucket to an external source?

### DISCUSSION THREAD

**Discussion by jake-edgenaut** [supabase/supabase#1839] [Questions] [answered]
Is there any way to do a 2-way file sync of a storage bucket to an external source such as OneDrive, NextCloud, etc.?

I could write a script of my own and run it on node, sure, but is there a more performant way, such as rsync?

Thanks for any ideas. I'm going to need to sync a storage bucket to a NextCloud instance for my project.

**Comment by inian** [comment, marked as answer]
Hi @ornithophile, I have created an issue to replicate to a user provided S3 bucket for now - https://github.com/supabase/storage-api/issues/27. From there you should be able to replicate to other sources since you will have raw access to the objects

**Reply by jake-edgenaut** [reply]
Thanks @inian , that would work I think.

Edit: NextCloud has an external storage plugin that allows mounting S3 buckets, so perfect.
